hole/intersection.go

```diff
@@ -113,33 +113,87 @@ func intersection() (args []string, out string) {
 	args = append(args, strconvbox(b2)+" "+strconvbox(b7))
 	outs = append(outs, "0")
 
-	//// generate 98 more random cases
+	//// generate 100 random cases
 	zeros := 0
 	nonZeros := 0
-	for zeros+nonZeros < 94 {
-		b1 = boxGen()
-		b2 = boxGen()
+	for zeros+nonZeros < 100 {
+		b1 := boxGen()
+		b2 := boxGen()
 		intersection := calculateIntersection(b1, b2)
 
-		// compute 90 non-zero cases and 8 zero ones
+		// compute 90 non-zero cases and 10 zero ones
 		if intersection > 0 && nonZeros < 90 {
-			args = append(args, strconvbox(b1)+" "+strconvbox(b2))
-			outs = append(outs, strconv.Itoa(intersection))
-			nonZeros++
-		} else if intersection == 0 && zeros < 4 {
-			args = append(args, strconvbox(b1)+" "+strconvbox(b2))
-			outs = append(outs, strconv.Itoa(intersection))
-			zeros++
+			args = append(args,
+				strconvbox(b1)+" "+strconvbox(b2))
+			outs = append(outs,
+				strconv.Itoa(intersection))
+			nonZeros += 1
+		} else if intersection == 0 && zeros < 10 {
+			args = append(args,
+				strconvbox(b1)+" "+strconvbox(b2))
+			outs = append(outs,
+				strconv.Itoa(intersection))
+			zeros += 1
 		}
 	}
 
-	// this should shuffle args and outputs in the same way
+	// 13x13 default side cases
+	// - |   |
+	// --|   |
+	// --|-  |
+	// --|---|
+	// --|---|-
+	//   |-  |
+	//   |-- |
```

Comment:
These two lines describe the same tests.
153 and 154 too

Reply:
indeed
good catch!